cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceTaskContext.java

```diff
@@ -107,6 +111,7 @@ public BasicMapReduceTaskContext(Program program,
     this.loggingContext = createLoggingContext(program.getId(), runId);
     this.spec = spec;
     this.plugins = Maps.newHashMap(program.getApplicationSpecification().getPlugins());
+    this.localizedResources = localizedResources;
```

Comment:
Better construct a `DefaultTaskLocalizationContext` and delegate to that so that all program types share the same underlying implementation.